dotnet/machinelearning
# [AutoML] CodeGen For AzureAttach

Transformer added:

ExtractPixel
NormalizeMapping
ResizeImage
ApplyOnnxModel
LabelMapping

Notify that those are transformers that not exist in AutoML

Make some refactor a bit in `AzureAttachCodeGenerator` so that it can expand to support Azure Attach smoothly plus more test friendly.

These are two E2E test for Azure CodeGen, 
[AzureCodeGeneratorTest](https://github.com/LittleLittleCloud/machinelearning/blob/da65aefadd40ea8cd35581a9fb44d1f3bc895096/test/Microsoft.ML.CodeGenerator.Tests/ApprovalTests/ConsoleCodeGeneratorTests.cs#L228)
[AzureImageCodeGeneratorTest](https://github.com/LittleLittleCloud/machinelearning/blob/da65aefadd40ea8cd35581a9fb44d1f3bc895096/test/Microsoft.ML.CodeGenerator.Tests/ApprovalTests/ConsoleCodeGeneratorTests.cs#L193)


See [UCI_Adult](https://github.com/LittleLittleCloud/UCI_Adult) for classification case

See [AzureImage](https://github.com/LittleLittleCloud/AzureImage) For Image Classification case

## Latest update (12/19/2019)

- add CSharp prefix to interface (ICSharpProjectGenerator, ICSharpSolutionGenerator.... etc)
- remove all partial class, (and IProjectFileGenerator). Create the ProjectFile class directly instead ( class CSharpProjectFile and class CSharpCodeFile) see [this](https://github.com/dotnet/machinelearning/blob/7eb3875ec163354c1d533ed91effa4f25a79aece/src/Microsoft.ML.CodeGenerator/CodeGenerator/CSharp/AzureCodeGenerator/AzureAttachConsoleAppCodeGenerator.cs#L39) for example
- implement `IProjectGenerator` interface to `ICSharpSolutionGenerator` so both Azure CodeGen and local CodeGen will have the same API
